### PR TITLE
virtual_network.iface_options: Add interface macvtap tests

### PR DESCRIPTION
**Pull request by rbian**
Add interface macvtap tests; update some fucntions.

### CODE REVIEW

File: libvirt/tests/cfg/virtual_network/iface_options.cfg [hunk 1]
```diff
@@ -64,11 +64,33 @@
...
+                    serial_login = "yes"
         - iface_unprivileged_user:
             unprivileged_user = "autotest"
             change_iface_options = "yes"
-            serail_login = "yes"
+            serial_login = "yes"
             iface_type = "bridge"
             iface_model = "virtio"
             iface_source = "{'bridge':'virbr0'}"
+        - iface_macvtap:
```
Comment (yumingfei) [inline libvirt/tests/cfg/virtual_network/iface_options.cfg]: IIRC, we already have some testcases about macvtap: tp-libvirt/libvirt/tests/src/macvtap.py
Do we need to merge them?

Comment (rbian) [inline libvirt/tests/cfg/virtual_network/iface_options.cfg]: The tests in macvtap.py depends on a remote host, the test environment is much more complex. There is no need to setup additional host or guest here, just check macvtap configuration by "ip link" and guest can get a valid ip address. I think the two implementations can keep in parallel.

**Comment by rbian** [comment]
Depends on autotest/virt-test#1997

**Comment by will-Do** [comment]
Tests are good enough, but could you please squash your patch series, then we can merge it
```
# ./run -t libvirt --test ydu-libvirt.virtual_network.iface_options.iface_macvtap -k --keep-image-between-test
SETUP: PASS (0.30 s)
DATA DIR: /root/Downloads/virt-test/shared/data
DEBUG LOG: /root/Downloads/virt-test/logs/run-2015-04-09-15.15.50/debug.log
TESTS: 8
(1/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_vhost.mode_bridge: PASS (102.32 s)
(2/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_vhost.mode_vepa: PASS (30.85 s)
(3/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_vhost.mode_private: PASS (31.09 s)
(4/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_vhost.mode_passthrough: PASS (30.77 s)
(5/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_qemu.mode_bridge: PASS (91.84 s)
(6/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_qemu.mode_vepa: PASS (30.78 s)
(7/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_qemu.mode_private: PASS (30.86 s)
(8/8) type_specific.ydu-libvirt.virtual_network.iface_options.iface_macvtap.driver_qemu.mode_passthrough: PASS (30.98 s)
TOTAL TIME: 379.55 s (06:19)
TESTS PASSED: 8
TESTS FAILED: 0
SUCCESS RATE: 100.00 %
[root@ydu-1 virt-test]# ./run -t libvirt --test ydu-libvirt.virtual_network.iface_options.iface_backend -k --keep-image-between-test
SETUP: PASS (0.30 s)
DATA DIR: /root/Downloads/virt-test/shared/data
DEBUG LOG: /root/Downloads/virt-test/logs/run-2015-04-09-15.23.03/debug.log
TESTS: 3
(1/3) type_specific.ydu-libvirt.virtual_network.iface_options.iface_backend.default_test: PASS (63.42 s)
(2/3) type_specific.ydu-libvirt.virtual_network.iface_options.iface_backend.specific_backend: PASS (26.08 s)
(3/3) type_specific.ydu-libvirt.virtual_network.iface_options.iface_backend.error_test: PASS (0.89 s)
TOTAL TIME: 90.43 s (01:30)
TESTS PASSED: 3
TESTS FAILED: 0
SUCCESS RATE: 100.00 %

**Comment by rbian** [comment]
Thanks @will-Do , updated.

**Pull request merged**